app/services/helpers/algolia/randomisation.rb

```diff
@@ -0,0 +1,24 @@
+module Helpers::Algolia
+  module Randomisation
+    include Index
+    include Pagination
+
+    def calculate_random_firm_ids(query:, seed:)
+      browse(query: query)
+        .map { |hit| hit['firm']['id'] }
+        .shuffle(random: Random.new(seed))
+        .each_slice(MAX_HITS_PER_PAGE)
+        .to_a
+    end
```

Comment:
my comment on this method is unprintable

Reply:
@margOnline can you clarify your views about this method, please? Then maybe we can work together on it and then having a better readable code.

Reply:
sorry, my comment was ironical

Reply:
I see. I don't see how an ironical comment can make us grow. I don't want to sound patronising but there is on mas-standards code of conduct [here](https://github.com/moneyadviceservice/mas-standards/blob/master/pull-request-code-of-conduct.md#reviewing-a-request) about

> Communicate which ideas you feel strongly about and those you don't.

So if you feel strongly that the code is not readable or simple or concise or precise, feel free to communicate which part needs improving and also try as best as possible to **offer alternative implementations**. In case you don't have alternative implementations is totally okay to ask to pair on it. Both of developers will learn from it and everybody will grow.

Also I asked clarification simply because I was following the code of conduct [here](https://github.com/moneyadviceservice/mas-standards/blob/master/pull-request-code-of-conduct.md#having-your-request-reviewed)

> If a review seems aggressive or angry or otherwise personal, consider if it is intended to be read that way and ask the person for clarification of intent, in person if possible.

Since we both not present and online at the same time, I decided to ask clarification here on the PR.